data/scripts/creaturescripts/player/idle_time.lua

```diff
@@ -0,0 +1,38 @@
+local config = {
+	maxIdleTime = 15 * 60 * 1000, -- 15 minutes
+	kickAfterWarn = 1 * 60 * 1000 -- 1 minute
+}
+
+-- register player event
+local event = CreatureEvent("RegisterIdleTime")
+
+function event.onLogin(player)
+	if not player:getGroup():getAccess() then
+		player:registerEvent("IdleTime")
+	end
+	return true
+end
+
+event:register()
+
+-- event to check player idle time
+event = CreatureEvent("IdleTime")
+
+function event.onThink(player, interval)
+	local tile = Tile(player:getPosition())
+	if not tile or tile:hasFlag(TILESTATE_NOLOGOUT) then
+		return true
+	end
+
+	local idleTime = player:getIdleTime() + interval
+	player:setIdleTime(idleTime)
+
+	if idleTime == config.maxIdleTime then
+		player:sendTextMessage(MESSAGE_STATUS_WARNING, "There was no variation in your behaviour for " .. (config.maxIdleTime / (60 * 1000)) .. " minutes. You will be disconnected in one minute if there is no change in your actions until then.")
```

Comment:
```suggestion
		player:sendTextMessage(MESSAGE_STATUS_WARNING, "There was no variation in your behaviour for " .. (config.maxIdleTime / (60 * 1000)) .. " minutes. You will be disconnected in " .. (config.kickAfterWarn / (60 * 1000)) .. " minutes if there is no change in your actions until then.")
```

Reply:
why is that? the kick happens exactly one minute after the message is sent, that is why it is "hard coded"

Reply:
I think it's because he's making it "configurable" now, so if somebody sets to kick after 2 minutes it will still send the "one minute" message...

Reply:
Exactly: he made it configurable through a table in the script
https://github.com/otland/forgottenserver/blob/5ccc6419e742623535aa938ec4673fc17ca37650/data/scripts/creaturescripts/player/idle_time.lua#L1-L4